modin/numpy/arr.py

```diff
@@ -0,0 +1,1488 @@
+        ErrorMessage.single_warning(
+            "Subclassing types is not currently supported in Modin. Defaulting to the same base dtype."
+        )
+    if keepdims:
+        ErrorMessage.single_warning(
+            "Modin does not yet support broadcasting between nested 1D arrays and 2D arrays."
+        )
+    if casting != "same_kind":
+        ErrorMessage.single_warning(
+            "Modin does not yet support the `casting` argument."
+        )
+    if not (
+        is_scalar(where) or (isinstance(where, array) and is_bool_dtype(where.dtype))
+    ):
+        if not isinstance(where, array):
+            raise NotImplementedError(
+                f"Modin only supports scalar or modin.numpy.array `where` parameter, not `where` parameter of type {type(where)}"
+            )
+        raise TypeError(
+            f"Cannot cast array data from {where.dtype} to dtype('bool') according to the rule 'safe'"
+        )
+
+
+def check_how_broadcast_to_output(arr_in: "array", arr_out: "array"):
+    if not isinstance(arr_out, array):
+        raise TypeError("return arrays must be of modin.numpy.array type.")
+    if arr_out._ndim == arr_in._ndim and arr_out.shape != arr_in.shape:
+        raise ValueError(
+            f"non-broadcastable output operand with shape {arr_out.shape} doesn't match the broadcast shape {arr_in.shape}"
+        )
+    elif arr_out._ndim == arr_in._ndim:
+        return "broadcastable"
+    elif arr_out._ndim == 1:
+        if prod(arr_in.shape) == arr_out.shape[0]:
+            return "flatten"
+        else:
+            raise ValueError(
+                f"non-broadcastable output operand with shape {arr_out.shape} doesn't match the broadcast shape {arr_in.shape}"
+            )
+    elif arr_in._ndim == 1:
+        if prod(arr_out.shape) == arr_in.shape[0]:
+            return "reshape"
+        else:
+            raise ValueError(
+                f"non-broadcastable output operand with shape {arr_out.shape} doesn't match the broadcast shape {arr_in.shape}"
+            )
+
+
+def fix_dtypes_and_determine_return(
+    query_compiler_in, _ndim, dtype=None, out=None, where=True
+):
+    if dtype is not None:
+        query_compiler_in = query_compiler_in.astype(
+            {col_name: dtype for col_name in query_compiler_in.columns}
+        )
+    result = array(_query_compiler=query_compiler_in, _ndim=_ndim)
+    if out is not None:
+        out = try_convert_from_interoperable_type(out)
+        broadcast_method = check_how_broadcast_to_output(result, out)
+        result._query_compiler = result._query_compiler.astype(
+            {col_name: out.dtype for col_name in result._query_compiler.columns}
+        )
+        if broadcast_method == "flatten":
+            result = result.flatten()
+        elif broadcast_method != "broadcastable":
+            # TODO(RehanSD): Replace this when reshape is implemented.
+            raise NotImplementedError("Reshape is currently not supported in Modin.")
```

Comment:
Why is a reshape needed here? Based on the logic of `check_how_to_broadcast_to_output`, wouldn't this "broadcastable" case only be hit if the shapes of `result` and `out` are identical?

Reply:
Right - so if its not equal to broadcastable, that means we need a reshape. The elif checks for inequality.

Reply:
Just checked and you're right - fixing!